pkgs/os-specific/linux/trusting-trust/tinycc/default.nix

```diff
@@ -0,0 +1,138 @@
+{ lib
+, mkKaemDerivation
+, fetchurl
+, callPackage
+, mes
+}:
+let
+  version = "unstable-2023-04-20";
+  src = builtins.fetchTarball {
```

Comment:
Working without access to the usual abstractions presents some fun challenges! `mescc-tools-extra` gives us access to `untar` and `ungz` but before that stage I can't think another way to efficiently fetch a source tree. `<nix/fetchurl.nix>` has an `unpack` argument but that is only able to handle `nar` archives

Reply:
Isn't fetchTarball specifically forbidden because it allows no hash and because of ``This function is not available if restricted evaluation mode is enabled.`` https://nixos.org/manual/nix/stable/language/builtins.html

Reply:
> Isn't fetchTarball specifically forbidden because it allows no hash

We can make sure we're using it with a hash, in this special case, if there's no other option.

> and because of `This function is not available if restricted evaluation mode is enabled.`

That one might be problematic.  It might be worth asking the Nix devs what the best way to do this would be.  Otherwise, maybe we can fetch a NAR with `<nix/fetchurl.nix>`.

Reply:
I think fetching a NAR could be a good option. To show it in action I've made a small repo to repackage and publish the minimal source files as NARs https://github.com/emilytrau/bootstrap-tools-nar-mirror/releases/tag/2023-04-25

Reply:
If you don't want to download a 3rd party nar it might also be possible to fetch every file from raw.githubusercontent.com. It would probably require manually generating the list and wouldn't be super clean though.

Reply:
We could upload it to tarballs.nixos.org maybe? That's where our current bootstrap tools live.

Reply:
IIRC @lovesegfault

Reply:
That's right, but it would happen as the very last step before landing this PR. 

Usually, contributors will host the tarball anywhere (e.g., GitHub, Web Archive, w/e) while they are iterating and going through review. Once the PR is ready, I will validate and upload the bootstrap tarball, and then we update the URL here to point to the right endpoint in `tarballs.nixos.org`, and we merge.

Reply:
Awesome! Once everything's ready to merge the script for building the tarball will be at [pkgs/os-specific/linux/minimal-bootstrap/stage0-posix/make-bootstrap-sources.nix](https://github.com/NixOS/nixpkgs/pull/227914/commits/cc8ee0925d9eb05a9a9197cbbe3b6b667b32bf02#diff-1b81d1b8139b102a4c33639890b2969906a72c1ff0a8d3450ff6f86c861bd297) :heart:

Reply:
Opened #232576